google/resource_google_project_default_service_accounts_test.go

```diff
@@ -0,0 +1,287 @@
+		}
+		for _, sa := range response.Accounts {
+			switch strings.ToLower(sa.DisplayName) {
+			case "compute engine default service account":
+				switch action {
+				case "DISABLE":
+					if !sa.Disabled {
+						return fmt.Errorf("compute engine default service account is not disabled, disable field is %t", sa.Disabled)
+					}
+				case "DELETE":
+					return fmt.Errorf("compute engine default service account is not deleted")
+				case "DEPRIVILEGE":
+					iamPolicy, err := config.NewResourceManagerClient(config.userAgent).Projects.GetIamPolicy(project, &cloudresourcemanager.GetIamPolicyRequest{
+						Options:         &cloudresourcemanager.GetPolicyOptions{},
+						ForceSendFields: []string{},
+						NullFields:      []string{},
+					}).Do()
+					if err != nil {
+						return fmt.Errorf("cannot get IAM policy on project %s: %v", project, err)
+					}
+					for _, bind := range iamPolicy.Bindings {
+						if bind.Role == "roles/editor" {
+							for _, member := range bind.Members {
+								if member == fmt.Sprintf("serviceAccount:%s", sa.Email) {
+									return fmt.Errorf("compute engine default service account is not deprivileged")
+								}
+							}
+						}
+					}
+					return nil
+				}
+			case "app engine default service account":
+				switch action {
+				case "DISABLE":
+					if !sa.Disabled {
+						return fmt.Errorf("app engine default service account is not disabled, disable field is %t", sa.Disabled)
+					}
+				case "DELETE":
+					return fmt.Errorf("app engine default service account is not deleted")
+				case "DEPRIVILEGE":
+					iamPolicy, err := config.NewResourceManagerClient(config.userAgent).Projects.GetIamPolicy(project, &cloudresourcemanager.GetIamPolicyRequest{
+						Options:         &cloudresourcemanager.GetPolicyOptions{},
+						ForceSendFields: []string{},
+						NullFields:      []string{},
+					}).Do()
+					if err != nil {
+						return fmt.Errorf("cannot get IAM policy on project %s: %v", project, err)
+					}
+					for _, bind := range iamPolicy.Bindings {
+						if bind.Role == "roles/editor" {
+							for _, member := range bind.Members {
+								if member == fmt.Sprintf("serviceAccount:%s", sa.Email) {
+									return fmt.Errorf("app engine default service account is not deprivileged")
+								}
+							}
+						}
+					}
+				}
+			default:
+				continue
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckGoogleProjectDefaultServiceAccountsRevert(t *testing.T, project, action string) func(s *terraform.State) error {
+	return func(s *terraform.State) error {
+		config := googleProviderConfig(t)
+		response, err := config.NewIamClient(config.userAgent).Projects.ServiceAccounts.List(prefixedProject(project)).Do()
+		if err != nil {
+			return fmt.Errorf("failed to list service accounts on project %q: %v", project, err)
+		}
+		for _, sa := range response.Accounts {
+			switch strings.ToLower(sa.DisplayName) {
```

Comment:
same here. Also, now that this has shown up a few times, it actually probably makes sense to have a separate `func isDefaultServiceAccount(displayName string) bool` or something like that

Reply:
I'm okay with creating a function for the main code, but my concern about using the same function for tests is that I broke the function I may break get false results from the tests. Thoughts?

Reply:
Wouldn't that be the exact same scenario as if you had two identical functions?

Reply:
yes, but I would not have two identical functions. We would have code function and test function with will eventually being the same, but allow me to test the logic in two steps. Update the code function, make sure the test fails, update the function code, make sure the tests pass... just thought I had.....

Reply:
That's fine! This is a pretty simple situation (just comparing strings) so I don't have a strong preference.